### ISSUE TITLE
New Octane Rule: `only-components-template-colocation`

### ISSUE THREAD

**Issue by patocallaghan** [ember-cli/eslint-plugin-ember#543]
Pulled from https://github.com/ember-cli/eslint-plugin-ember/issues/533 which proposes a new set of rules for Octane related best practices

Only use [Component Templates Colocation](https://github.com/emberjs/rfcs/blob/master/text/0481-component-templates-co-location.md)

```js
// ### Good
app/components/my-component.js
app/components/my-component.hbs
app/components/my-component/index.js
app/components/my-component/index.hbs
app/components/nested/other-component.js
app/components/nested/other-component.hbs
app/components/nested/other-component/index.js
app/components/nested/other-component/index.hbs

// ### Bad
app/components/my-component.js
app/templates/components/my-component.hbs
app/components/nested/other-component.js
app/templates/components/nested/other-component.hbs
```

If anyone would like to implement this please comment that you'll take it and work away.

**Comment by patocallaghan** [comment]
@rwjblue @Turbo87 I'm a bit fuzzier on the good/bad examples of this one. I'm also not sure on the implementation details of this. Do we still use ember-template-recast? Is there prior art here in other templates?

**Comment by rwjblue** [comment]
Might need another "linter" for filesystem things :thinking:

**Comment by patocallaghan** [comment]
@rwjblue Actually thinking about it this rule probably does make as much here? Considering it's for _template_ colocation it probably would fit better in the ember-template-lint repo?

If you agree I'll close this issue here and move it over there.

**Comment by rwjblue** [comment]
Seems good to me...

**Comment by patocallaghan** [comment]
Closing this issue here and I'll re-open on ember-template-lint

**Comment by patocallaghan** [comment]
Re-opened in https://github.com/ember-template-lint/ember-template-lint/issues/863
